pkg/clustermesh/metrics.go

```diff
@@ -22,25 +22,25 @@ type Metrics struct {
 func NewMetrics() Metrics {
 	return Metrics{
 		TotalNodes: metric.NewGaugeVec(metric.GaugeOpts{
-			ConfigName: metrics.Namespace + "_" + subsystem + "_remote_cluster_nodes",
+			ConfigName: metrics.Namespace + "_" + metrics.SubsystemClusterMesh + "_remote_cluster_nodes",
```

Comment:
While you are here, could you drop the `ConfigName` field? It used to be necessary back then, but it is now [automatically computed if unset](https://github.com/cilium/cilium/blob/5ce5669732e96dc41dc02ec64ef66c1cd3f84fef/pkg/metrics/metric/metric.go#L275-L280).